File: import_i18n_from_csv.js
#!/usr/bin/env node
/**
 * Import translations from a CSV file into JSON language files.
 * CSV must have columns: key, fr, en, es, ar, zh
 * Usage:
 *   node tools/import_i18n_from_csv.js path/to/protegemoi_i18n_keys.csv
 */
const fs = require('fs');
const path = require('path');

function parseCSV(content){
  const [headerLine, ...lines] = content.split(/\r?\n/).filter(Boolean);
  const headers = headerLine.split(',').map(h => h.trim());
  return lines.map(line => {
    // naive CSV split; ok for simple content without commas in fields
    const cells = line.split(',').map(c => c.trim());
    const obj = {};
    headers.forEach((h, i) => obj[h] = cells[i] || '');
    return obj;
  });
}

function main(csvPath){
  if(!fs.existsSync(csvPath)){
    console.error('[Error] CSV not found:', csvPath);
    process.exit(1);
  }
  const projectRoot = path.resolve(__dirname, '..');
  const i18nDir = path.join(projectRoot, 'assets', 'i18n');
  const langs = ['fr','en','es','ar','zh'];

  const dicts = {};
  langs.forEach(lang => {
    const f = path.join(i18nDir, `${lang}.json`);
    try{
      dicts[lang] = JSON.parse(fs.readFileSync(f, 'utf8'));
    }catch{
      dicts[lang] = {};
    }
  });

  const csv = fs.readFileSync(csvPath, 'utf8');
  const rows = parseCSV(csv);
  const required = ['key','fr','en','es','ar','zh'];
  if(!required.every(k => rows.length === 0 || Object.keys(rows[0]).includes(k))){
    console.error('[Error] CSV must include columns:', required.join(', '));
    process.exit(1);
  }
  let count = 0;
  rows.forEach(row => {
    const key = (row.key || '').trim();
    if(!key) return;
    langs.forEach(lang => {
      dicts[lang][key] = (row[lang] || '');
    });
    count++;
  });

  langs.forEach(lang => {
    const out = path.join(i18nDir, `${lang}.json`);
    fs.writeFileSync(out, JSON.stringify(dicts[lang], null, 2), 'utf8');
  });

  console.log(`[OK] Imported translations for ${count} keys into ${i18nDir}`);
}

if(require.main === module){
  const csvPath = process.argv[2];
  if(!csvPath){
    console.log('Usage: node tools/import_i18n_from_csv.js path/to/protegemoi_i18n_keys.csv');
    process.exit(1);
  }
  main(csvPath);
}

File: import_i18n_from_csv.py
#!/usr/bin/env python3
"""
Import translations from a CSV file into JSON language files.
CSV must have columns: key, fr, en, es, ar, zh
Usage:
    python tools/import_i18n_from_csv.py path/to/protegemoi_i18n_keys.csv
"""
import sys, json, csv
from pathlib import Path

def main(csv_path: str):
    csv_file = Path(csv_path)
    if not csv_file.exists():
        print(f"[Error] CSV not found: {csv_file}")
        sys.exit(1)

    project_root = Path(__file__).resolve().parents[1]
    i18n_dir = project_root / "assets" / "i18n"
    langs = ["fr","en","es","ar","zh"]

    # Load current JSONs (or create)
    dicts = {}
    for lang in langs:
        f = i18n_dir / f"{lang}.json"
        if f.exists():
            try:
                dicts[lang] = json.loads(f.read_text(encoding="utf-8"))
            except Exception:
                dicts[lang] = {}
        else:
            dicts[lang] = {}

    # Read CSV and apply updates
    with csv_file.open("r", encoding="utf-8") as f:
        reader = csv.DictReader(f)
        required_cols = {"key","fr","en","es","ar","zh"}
        if not required_cols.issubset(set(reader.fieldnames or [])):
            print(f"[Error] CSV must include columns: {', '.join(sorted(required_cols))}")
            sys.exit(1)
        count = 0
        for row in reader:
            key = (row.get("key") or "").strip()
            if not key:
                continue
            for lang in langs:
                val = row.get(lang, "")
                if val is None:
                    val = ""
                dicts[lang][key] = val
            count += 1

    # Write back JSON files with pretty formatting
    for lang in langs:
        out = i18n_dir / f"{lang}.json"
        out.write_text(json.dumps(dicts[lang], ensure_ascii=False, indent=2), encoding="utf-8")

    print(f"[OK] Imported translations for {count} keys into {i18n_dir}")

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: python tools/import_i18n_from_csv.py path/to/protegemoi_i18n_keys.csv")
        sys.exit(1)
    main(sys.argv[1])
